services/catalog/src/simcore_service_catalog/utils/pools.py

```diff
@@ -0,0 +1,13 @@
+from contextlib import contextmanager
+from concurrent.futures import ProcessPoolExecutor
+
+
+@contextmanager
+def non_blocking_process_pool_executor(**kwargs) -> ProcessPoolExecutor:
+    """
+    Avoids default context manger behavior which calls
+    shutdown with wait=True an blocks.
+    """
+    pool = ProcessPoolExecutor(**kwargs)  # pylint: disable=consider-using-with
+    yield pool
```

Comment:
I do not see how this differs from the context manager. 
IMO we should have a single ProcessPoolExecutor application-wide, similar to the http client. 
It can be initialized lazily (see get_client_application ) or during setup ... and should be closed upon app shutdown.
I would reuse my suggestion in ``setup_monitoring``

Reply:
I've looked into this and where a contextmanger was used I've decided against having a shared pool. That's how we have the current implementation.

Reply:
But you have a shared pool in the web-server app and not here?

Reply:
Yes, this was a pool limited to 2 processes each time a `GET /services/list_services` request is submitted.
Not being sure about how often it's called, preserving the current pattern will not alter the present behaviour.